lib/jxl/headers.cc

```diff
@@ -63,15 +63,15 @@ Status SizeHeader::Set(size_t xsize64, size_t ysize64) {
   const uint32_t xsize32 = static_cast<uint32_t>(xsize64);
   const uint32_t ysize32 = static_cast<uint32_t>(ysize64);
   if (xsize64 == 0 || ysize64 == 0) return JXL_FAILURE("Empty image");
-  small_ = xsize64 <= 256 && ysize64 <= 256 && (xsize64 % kBlockDim) == 0 &&
-           (ysize64 % kBlockDim) == 0;
+  ratio_ = FindAspectRatio(xsize32, ysize32);
+  small_ = ysize64 <= 256 && (ysize64 % kBlockDim) == 0 &&
+           (ratio_ > 0 || (xsize64 <= 256 && (xsize64 % kBlockDim) == 0));
```

Comment:
should we use != 0 instead? I think it's clearer.

Reply:
I am always tempted to write just `ratio_` in cases like this, but I guess that's bad style :)

In my head there's no difference between > 0 and != 0 for a uint, but if != 0 looks clearer to you, let's do that.